deltachat-ffi/src/providers.rs

```diff
@@ -0,0 +1,93 @@
+use deltachat::dc_tools::{as_str, dc_strdup, StrExt};
+extern crate deltachat_provider_overview;
+use deltachat_provider_overview::StatusState;
+use std::ptr;
+
+#[no_mangle]
+pub type dc_provider = deltachat_provider_overview::Provider;
+
+#[no_mangle]
+pub unsafe extern "C" fn dc_get_json_provider_info_from_email(
+    email: *const libc::c_char,
+) -> *mut libc::c_char {
+    let option = deltachat_provider_overview::get_provider_info(
+        deltachat_provider_overview::get_domain_from_email(as_str(email)),
+    );
+    if let Some(provider) = option {
+        serde_json::to_string(provider.0).unwrap().strdup()
+    } else {
+        dc_strdup(ptr::null())
```

Comment:
aka `"".strdup()`.  Reason I suggest this is because with `dc_strdup()` I need to think about what it does compared to `dc_strdup_keep_null()` and which one this actually is and what it does.

However I question if it is the right approach to return an empty string here.  At least in python when i try and parse an empty string as json I get an error, suggesting this isn't actually valid json.  So maybe it's better to return NULL as error code here?

Reply:
@Simon-Laux Sounds convincing, shall I change this to return NULL?

Reply:
I'm not sure how this works with pointers to strings. A change here maybe needs also a change in the python bindings, I don't know so try it @flub has more knowledge here than me.